projects/plugins/boost/app/assets/src/js/lib/stores/page-cache.ts

```diff
@@ -69,6 +69,41 @@ export function usePageCacheSetup() {
 	return pageCacheSetup;
 }
 
+/**
+ * Hook which creates a callable action for checking Page Cache setup.
+ */
+export function usePageCacheSetupCheck() {
+	const [ , pageCacheErrorMutation ] = usePageCacheError();
+	const setError = pageCacheErrorMutation.mutate;
+
+	const pageCacheSetup = useDataSyncAction( {
+		namespace: 'jetpack_boost_ds',
+		key: 'page_cache',
+		action_name: 'check-setup',
+		schema: {
+			state: PageCache,
+			action_request: z.void(),
+			action_response: PageCacheError.or( z.literal( true ) ),
+		},
+		mutationOptions: {
+			onError: error => {
+				if ( error instanceof DataSyncError ) {
+					return setError( error.info() );
+				}
+				const standardizedError = standardizeError( error );
+				setError( {
+					code: 'unknown_error',
+					message: standardizedError.message || __( 'Unknown error occurred.', 'jetpack-boost' ),
+				} );
+			},
+			onSuccess: () => {
+				setError( null );
+			},
+		},
+	} );
```

Comment:
This is a whole lot of repeat code from the function above it. Can these be combined?